org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest2.java

```diff
@@ -343,11 +356,35 @@ private static String mergeIntoCommaSeparatedList(String bundle, String oldBundl
 	}
 
 	private static String bundleName(String bundle) {
-		return bundle.contains(";") ? bundle.split(";")[0] : bundle;
+		if (!bundle.contains(";"))
```

Comment:
We do split this multiple times. Again: Not relevant for performance, but a result object that encapsulates the bunde name, suffix and version would be a cleaner solution. It could also expose a merge method and a reasonable toString()

```
static class RequiredBundle {
  String name, version, suffix // I don't like the name suffix
  RequiredBundle merge(String entry} { .. }
  String toManifest() {
     return ...
  }
}

```

Reply:
Will extract classes "Bundle" and "BundleList".